iceoryx_posh/include/iceoryx_posh/internal/popo/server_impl.inl

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) 2022 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef IOX_POSH_POPO_SERVER_IMPL_INL
+#define IOX_POSH_POPO_SERVER_IMPL_INL
+
+#include "iceoryx_posh/internal/popo/server_impl.hpp"
+
+namespace iox
+{
+namespace popo
+{
+template <typename Req, typename Res, typename BaseServerT>
+inline ServerImpl<Req, Res, BaseServerT>::ServerImpl(const capro::ServiceDescription& service,
+                                                     const ServerOptions& serverOptions) noexcept
+    : BaseServerT(service, serverOptions)
+{
+}
+
+template <typename Req, typename Res, typename BaseServerT>
+cxx::expected<Request<const Req>, ServerRequestResult> ServerImpl<Req, Res, BaseServerT>::take() noexcept
+{
+    auto result = port().getRequest();
+    if (result.has_error())
+    {
+        return cxx::error<ServerRequestResult>(result.get_error());
+    }
+    auto requestHeader = result.value();
+    auto payload = mepoo::ChunkHeader::fromUserHeader(requestHeader)->userPayload();
+    auto request = cxx::unique_ptr<const Req>(static_cast<const Req*>(payload), m_requestDeleter);
+    return cxx::success<Request<const Req>>(Request<const Req>{std::move(request)});
+}
+
+template <typename Req, typename Res, typename BaseServerT>
+cxx::expected<Response<Res>, AllocationError>
+ServerImpl<Req, Res, BaseServerT>::loanUninitialized(const Request<const Req>& request) noexcept
+{
+    const auto* requestHeader = &request.getRequestHeader();
+    auto result = port().allocateResponse(requestHeader, sizeof(Req), alignof(Req));
+    if (result.has_error())
+    {
+        return cxx::error<AllocationError>(result.get_error());
+    }
+    else
```

Comment:
Could you please perform just an early return and move the code inside the else branch outside and remove the `else`